api/v1alpha1/bundle_types.go

```diff
@@ -95,6 +100,16 @@ type GitRef struct {
 	Commit string `json:"commit,omitempty"`
 }
 
+type Authorization struct {
+	// Type defines the kind of authorization method being used.
+	AuthorizationType string `json:"authorizationType,omitempty"`
```

Comment:
I think we only want `AuthorizationType` if we actually have a union (something like):
```go
type Authorization struct {
	AuthorizationType string `json:"authorizationType,omitempty"`
	Secret corev1.LocalObjectReference `json:"secret,omitempty"`
	HTTP *HTTPAuthorization `json:"http,omitempty"`
	SSH *SSHAuthorization `json:"ssh,omitempty"`
}

type HTTPAuthorization struct {
	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
}
...
```

I'm torn though. Another union type seems like more complexity for us to manage. And even if we're doing https, it's only necessary to include the HTTPAuthorization struct if you need to enable InsecureSkipVerify.

Reply:
Maybe for now, we keep things simple and:
1. Drop `AuthorizationType`
2. Document in GoDoc that `InsecureSkipVerify` is only used when cloning repos using the `https` scheme.